### DISCUSSION TITLE
Report | Week 06

### DISCUSSION THREAD

**Discussion by tech-n1c** [SE-TINF22B6/time2shine#35] [Show and tell]
## Overview
This week we created our CRC and ER diagrams. 

---

### ER diagram
We created our ER diagram using the Diagrams tool of JetBrains DataGrip Software. It can be used for accessing many dfferent  databases (writing SQL queries etc.). On the other hand, you can also visualize your databases in form of ER diagrams. This way, we didn't have to fully draw the diagram ourselves and also hat the database setup in one relatively simple step. So we'd definitely recommend that to other groups as well, because it has a very nice and beginner-friendly User Interface.

![image](https://raw.githubusercontent.com/SE-TINF22B6/time2shine/main/Documentation/ER-Diagram.png)

---

### Class Responsibility Collaboration (CRC) Diagram

![image](https://raw.githubusercontent.com/SE-TINF22B6/time2shine/c0febea8cdf6e62de23cd6ecf6f96cd46d248748/Documentation/CRC.drawio.svg)

#### Understanding the Key Classes
- Account represents a user account in the game. It has attributes such as username, email, highscore, friends, and AccountID. It also has methods for adding friends and joining games.
- Game Room represents a room where a game is played. It has attributes such as RoomID, Players, and GameType. It also has methods for refreshing the highscores table.
- Game Type represents the type of game being played. It has an attribute for the name of the game.
- Player represents a player in a game. It has attributes such as AccountID and SetOfRules. It also has methods for checking scores and drawing cards.

#### The relationships between the classes are as follows:

- Account has a many-to-many relationship with Game Room, meaning that an account can join multiple game rooms, and a game room can have multiple accounts joined.
- Game Room has a one-to-one relationship with Game Type, meaning that a game room can only have one game type, and a game type can only be associated with one game room.
- The CRC diagram also shows the following collaborations between the classes:

#### Account uses Game Room to join games.
- Game Room uses Player to track the players in the game.
- Player uses Game Room to check the game type and highscores.
- Game Room uses Game Type to determine the rules of the game.
- Player uses Game Room to draw cards and check scores.
- This CRC diagram provides a high-level overview of the relationships between the key classes in a game. It can be used to identify the responsibilities of each class and to ensure that the classes collaborate effectively.

**Comment by denniskp** [comment]
Dear Team time2shine,
Your designs for the diagrams are very clearly arranged. Also, your documentation for these is very detailed, which makes it easy to understand how your database is structured in view of dependencies, associations, compositions and aggregations between classes.
You've already achieved a lot in this sprint, we are excited to see what's still to come!

**Comment by derrrobin** [comment]
Hey, 
I really like the design of your class diagramm. Also I for me you did a great job with the explenation. perhaps it would also be good for the reader to have a more detailed description of your ERR diagram. Your CookPit Team

**Comment by Shiny1708** [comment]
I like the concept that you showed with your diagram. One thing you have to be careful with is not sharing too much information with the highscore board about the users so account security will not be compromised.

**Comment by iscodeminister** [comment]
Dear Time2Shine Team,
Your diagram designs with the software are visually appealing to me. It's very clear to know, and the descriptions are also. I'm looking forward to your future updates.
